static/js/message_viewport.js

```diff
@@ -48,19 +48,24 @@ export function message_viewport_info() {
     // Return a structure that tells us details of the viewport
     // accounting for fixed elements like the top navbar.
     //
-    // The message_header is NOT considered to be part of the visible
+    // Sticky message_header is NOT considered to be part of the visible
     // message pane, which should make sense for callers, who will
     // generally be concerned about whether actual message content is
     // visible.
 
     const res = {};
 
-    const $element_just_above_us = $(".floating_recipient");
+    const $element_just_above_us = $("#navbar-container .header");
     const $element_just_below_us = $("#compose");
 
     res.visible_top =
         $element_just_above_us.offset().top + $element_just_above_us.safeOuterHeight();
 
+    const $sticky_header = $(".sticky_header");
+    if ($sticky_header.length) {
+        res.visible_top += $sticky_header.safeOuterHeight();
+    }
```

Comment:
Just to make sure I understand, this calculation and the below one in this file are both the same as before -- the difference is just that previously, there would always be a `.floating_recipient` element that was possibly not visible and thus had 0 height, and now there sometimes is no such element?

Reply:
Yes.